src/common/dav/tx.c

```diff
@@ -452,18 +452,21 @@ tx_create_wal_entry(struct ulog_entry_base *e, void *arg,
 int
 lw_tx_begin(dav_obj_t *pop)
 {
-	struct umem_wal_tx	*utx = pop->do_utx;
+	struct umem_wal_tx	*utx = NULL;
 	int			 rc;
 
-	D_ASSERT(utx == NULL);
```

Comment:
Could you explain a bit why this assert is removed?

Reply:
umem/dav_reserve() will create a wal entries without actually commiting it. If atomic api is called post this we will end up hitting this assert

Reply:
Sorry, I forgot why reserve needs to generate WAL entries, but can we guarantee that the WAL entries for reserve are committed by the publish transaction? For example:
1. ULT_A does reserve;
2. ULT_B start & commit transaction B (or call atomic API B);
3. ULT_A commits transaction A to publish reservation;

The reserve won't be mistakenly committed by B, right?

Reply:
reserve will occasionally create wal entries if the allocation requires new chunk or zone to be created. We do not have undo mechanism for chunk/zone creation and this change needs to be committed along with the next commit that happens to the vos file. The commit of actual allocation block will happen only when ULT_A commits its transaction. In the above example when ULT_B does commit, it commits the changes related to chunk and zone creation and when ULT_A does tx_publish the redo entry for the allocation block within the chunk will be created.

Reply:
I see, thanks.